planning/behavior_path_planner/autoware_behavior_path_planner_common/test/test_parking_departure_utils.cpp

```diff
@@ -0,0 +1,315 @@
+#include "autoware/behavior_path_planner_common/utils/path_safety_checker/path_safety_checker_parameters.hpp"
+
+#include <ament_index_cpp/get_package_share_directory.hpp>
+#include <autoware_test_utils/autoware_test_utils.hpp>
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+#include <lanelet2_core/Forward.h>
+
+#include <cstddef>
+
+constexpr double epsilon = 1e-6;
+
+using autoware::behavior_path_planner::PlannerData;
+using autoware_planning_msgs::msg::Trajectory;
+using tier4_planning_msgs::msg::PathPointWithLaneId;
+using tier4_planning_msgs::msg::PathWithLaneId;
+
+using autoware::test_utils::generateTrajectory;
+
+PathWithLaneId trajectory_to_path_with_lane_id(const Trajectory & trajectory)
+{
+  PathWithLaneId path_with_lane_id;
+  PathPointWithLaneId path_point_with_lane_id;
+  for (const auto & point : trajectory.points) {
+    path_point_with_lane_id.point.pose = point.pose;
+    path_point_with_lane_id.point.lateral_velocity_mps = point.lateral_velocity_mps;
+    path_point_with_lane_id.point.longitudinal_velocity_mps = point.longitudinal_velocity_mps;
+    path_point_with_lane_id.point.heading_rate_rps = point.heading_rate_rps;
+    path_with_lane_id.points.push_back(path_point_with_lane_id);
+  }
+  return path_with_lane_id;
+}
+
+TEST(BehaviorPathPlanningParkingDepartureUtil, calcFeasibleDecelDistance)
+{
+  using autoware::behavior_path_planner::utils::parking_departure::calcFeasibleDecelDistance;
+
+  auto data = std::make_shared<PlannerData>();
+  double velocity = 2.0;
+  double acceleration = 1.0;
+  double acceleration_limit = 2.0;
+  double jerk_limit = 1.0;
+  double target_velocity = 3.0;
+
+  auto odometry = std::make_shared<nav_msgs::msg::Odometry>();
+  odometry->pose.pose = autoware::test_utils::createPose(0.0, 0.0, 0.0, 0.0, 0.0, 0.0);
+  odometry->twist.twist.linear.x = velocity;
+  auto accel_with_covariance =
+    std::make_shared<autoware::behavior_path_planner::AccelWithCovarianceStamped>();
+  accel_with_covariance->accel.accel.linear.x = acceleration;
+  data->self_odometry = odometry;
+  data->self_acceleration = accel_with_covariance;
+  auto planner_data = std::static_pointer_cast<const PlannerData>(data);
+
+  // condition: current velocity is slower than target velocity
+  auto distance =
+    calcFeasibleDecelDistance(planner_data, acceleration_limit, jerk_limit, target_velocity);
+  ASSERT_TRUE(distance.has_value());
+  EXPECT_NEAR(distance.value(), 0.0, epsilon);
+
+  // condition: calculates deceleration distance
+  velocity = 5.0;
+  odometry->twist.twist.linear.x = velocity;
+  data->self_odometry = odometry;
+  planner_data = std::static_pointer_cast<const PlannerData>(data);
+  distance =
+    calcFeasibleDecelDistance(planner_data, acceleration_limit, jerk_limit, target_velocity);
+  ASSERT_TRUE(distance.has_value());
+  EXPECT_NEAR(distance.value(), 18.7730133, epsilon);
+
+  // condition: nota valid condition
```

Comment:
```suggestion
  // condition: not valid condition
```
typo?

Reply:
Fixed in 38cf63225521af5be868738554a594a9b1b8795e